client/commonFramework/add-loop-protect.js

```diff
@@ -4,13 +4,16 @@ window.common = (function(global) {
     common = { init: [] }
   } = global;
 
-  loopProtect.hit = function hit(line) {
-     var err = 'Error: Exiting potential infinite loop at line ' +
-     line +
-     '. To disable loop protection, write: \n\\/\\/ noprotect\nas the first' +
-     'line. Beware that if you do have an infinite loop in your code' +
-     'this will crash your browser.';
-    console.error(err);
+  loopProtect.hit = function(line) {
+    var msg = 'Potential infinite loop at line ' +
+      line +
+      '. To disable loop protection, write:' +
+      ' \\n\\/\\/ noprotect\\nas the first' +
+      ' line. Beware that if you do have an infinite loop in your code' +
+      ' this will crash your browser.';
+    window.__err = new Error(msg);
```

Comment:
I left the console.error because as far as I can tell, in certain cases we are not displaying the error. In the original repro for Drop It, the error only happens on the fourth test and is not surfaced to the user. Or maybe I'm just not understanding the interface.